sdk/helper/tokenutil/root/encode.go

```diff
@@ -0,0 +1,48 @@
+package root
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/helper/pgpkeys"
+	"github.com/hashicorp/vault/sdk/helper/xor"
+)
+
+var (
+	// ErrNoTokenProvided is returned when an empty string is provided as an error
+	ErrNoTokenProvided = errors.New("no token provided")
+	// ErrNoOTPorPGPKeyProvided is returned when a non-empty token is provided but not OTP or PGP key
+	ErrNoOTPorPGPKeyProvided = errors.New("no otp or pgp key provided")
+)
+
+// EncodeToken gets a token and an OTP or a PGP Key and encodes it. If no OTP or PGP key is
+// provided, an error will be returned. If an OTP is provided, it must have the same length as the token
+func EncodeToken(token, otp, pgpKey string, cleanupFunc func()) (string, error) {
+	if len(token) == 0 {
+		cleanupFunc()
+		return "", ErrNoTokenProvided
+	}
+	// Get the encoded value first so that if there is an error we don't create
+	// the root token.
+	if len(otp) > 0 {
```

Comment:
Not sure why you replaced the original switch statement with this if-else statement, but we tend to prefer the former; many Go coders eschew `else` altogether in favour of switches and early returns.

Reply:
I'm not a huge fan of switch statements tbh. Out of curiosity I checked my company's codebase which is 90% Go and there isn't much love for switches in it either. OTOH I don't have a strong opinion on this either so I will switch to switch (pun intended)

Reply:
Fair, I wouldn't normally read someone's PR and say "could you turn this if/else into a switch", the only reason I commented here was because you were turning an existing switch into an if/else.  No big deal either way.